### PR TITLE
*: Use schema errors in ddl

### PR DESCRIPTION
**Pull request by shenli**
Remove ddl errors and use schema errors.

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: stmt/stmts/drop.go [hunk 1]
```diff
@@ -139,7 +138,7 @@ func (s *DropTableStmt) Exec(ctx context.Context) (rset.Recordset, error) {
 		}
 
 		err = sessionctx.GetDomain(ctx).DDL().DropTable(ctx, fullti)
-		if terror.ErrorEqual(err, ddl.ErrNotExists) || infoschema.DatabaseNotExists.Equal(err) {
+		if infoschema.DatabaseNotExists.Equal(err) {
```
Comment (qiuyesuifeng) [inline stmt/stmts/drop.go]: Here should be TableNotExists?

File: tidb-server/server/server_test.go [hunk 2]
```diff
@@ -232,12 +232,17 @@ func runTestErrorCode(c *C) {
...
 		txn2, err := dbt.db.Begin()
 		c.Assert(err, IsNil)
 		_, err = txn2.Exec("use db_not_exists;")
 		checkErrorCode(c, err, tmysql.ErrBadDb)
 		_, err = txn2.Exec("select * from tbl_not_exists;")
 		checkErrorCode(c, err, tmysql.ErrNoSuchTable)
+		_, err = txn2.Exec("create database test;")
+		checkErrorCode(c, err, tmysql.ErrDbCreateExists)
+		_, err = txn2.Exec("create table test (c int);")
+		checkErrorCode(c, err, tmysql.ErrTableExists)
```
Comment (qiuyesuifeng) [inline tidb-server/server/server_test.go]: Are there any tests for drop table and drop db?

Comment (shenli) [inline tidb-server/server/server_test.go]: PTAL @qiuyesuifeng

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**